x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -21,6 +21,15 @@ pub enum ExternalCallError {
     OutOfFuel = 2,
 }
 
+#[derive(Error, Debug, BorshDeserialize)]
+#[repr(u8)]
+#[non_exhaustive]
+#[borsh(use_discriminant = true)]
+pub enum DeployError {
+    #[error("an error happened during the deploy")]
+    DeployFailure = 0,
+}
```

Comment:
I see little benefit in having a separate `DeployError`, it might be better to merge the `ExternalCallError` and `DeployError` into a `ProgramError`

Reply:
I'd rather have more specific errors for the calls if possible.  It gives an easy way to see what errors are applicable.

Reply:
what I meant is that it's a little bit weird to have a separate error enum while we use `ExternalCallError` for all other faillible program operations because then we would need to import two different error enums, just thought about having
```rust
#[derive(Error, Debug, BorshDeserialize)]
#[repr(u8)]
#[non_exhaustive]
#[borsh(use_discriminant = true)]
pub enum ExternalCallError {
    #[error("an error happened during the execution")]
    ExecutionFailure = 0,
    #[error("the call panicked")]
    CallPanicked = 1,
    #[error("not enough fuel to cover the execution")]
    OutOfFuel = 2,
    #[error("an error happened during the deploy")]
    DeployFailure = 3,
}
```

But since `ExternalCallError` isn't really a good name anymore because of the `DeployFailure`, I suggested `ProgramError`.

Reply:
The reason this error is called `ExternalCallError` is because it should only be returned on external calls. I'm not sure that a `DeployFailure` deserves its own category when making an external call. What are the circumstances in which a deployment can fail? You could run out of fuel, certainly... but that should be an `OutOfFuel` error. 

I don't think a deployment failure should be recoverable. Either that, or we need to give a better reason as to why the deployment failed.